frontend/__tests__/components/graphs/helpers.spec.ts

```diff
@@ -0,0 +1,62 @@
+import { getPrometheusURL, PrometheusEndpoint } from '@console/internal/components/graphs/helpers';
+
+describe('getPrometheusURL()', () => {
+  it('should build a Prometheus /api/v1/query URL', () => {
+    const url = getPrometheusURL({
+      namespace: 'test-namespace',
+      endpoint: PrometheusEndpoint.QUERY,
+      query: 'test-query',
+    });
+
+    // Check that url contains all expected params, in no particular order
+    expect(url).toContain('/api/v1/query?');
+    expect(url).toContain('namespace=test-namespace');
+    expect(url).toContain('query=test-query');
```

Comment:
I'd use `URL` and `URLSearchParams` to verify the URL is correct instead of substring matching.

Reply:
Changed